solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/PropertyList/PropertyListViewModel.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  PropertyListViewModel.swift
+//  RealEstateApp
+//
+//  Created by Caio Alcântara on 23/04/22.
+//
+
+import Foundation
+
+protocol PropertyListViewModelProtocol: AnyObject {
+    var didFetch: ([Property]) -> () { get set }
+    func fetchListProperty()
+}
+
+final class PropertyListViewModel : PropertyListViewModelProtocol {
+    var didFetch: ([Property]) -> () = { _ in }
+    private let apiClient: RealEstateAPIPropertyProtocol
+    
+    init(apiClient: RealEstateAPIPropertyProtocol) {
+        self.apiClient = apiClient
+    }
+    
+    func fetchListProperty() {
```

Comment:
Utilizar o escaping aqui, por segurança, o didFetch esta atribuido como string, sendo uma closure, e sendo chamada dentro de outra. Pode ser que tenhamos aqui um retain cycle, mesmo com uso do weak self.
Podemos usar:
1. Escaping no fetchListProperty
2. Implementar Bindable: https://www.swiftbysundell.com/articles/bindable-values-in-swift/

Se seguir a opcao 2, aprendera sobre o MVVM utilizando binds sem uso do Combine/RX e afins, de forma mais simplificada.